File: modules/core/src/main/java/com/eventhorizon/pulsar/core/blueprint/Blueprint.java
package com.eventhorizon.pulsar.core.blueprint;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;
import java.util.Map;

public record Blueprint(
        String id,
        String name,
        @JsonProperty("blueprint_type") BlueprintType type,
        List<Volume> volumes,
        Map<String, String> environment
) {}

File: modules/core/src/main/kotlin/com/eventhorizon/pulsar/core/blueprint/Blueprint.kt
package com.eventhorizon.pulsar.core.blueprint

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Blueprint(
    val id: String,
    val name: String,
    @SerialName("blueprint_type") val type: BlueprintType,
    val volumes: List<Volume>,
    val environment: Map<String, String>
)

@Serializable
enum class BlueprintType {
    STANDALONE,
    HYTALE
}

@Serializable
data class Volume(
    val id: String,
    @SerialName("target_folder") val targetFolder: String
)